rio-code/src/main/java/frc/robot/subsystems/RobotMode.java

```diff
@@ -0,0 +1,200 @@
+        SwerveYSupplier = () -> 0.0;
+        SwerveRotationSupplier = () -> 0.0;
+
+        alignWithCoralStation = new AlignWithCoralStation();
+        alignWithLeftReef = new AlignWithLeftReef();
+        alignWithRightReef = new AlignWithRightReef();
+        controllerDrive = new ControllerDrive();
+    
+        coralStation = new CoralStation();
+        transitPose = new TransitPose();
+        transit = new Transit();
+        coralStationPose = new CoralStationPose();
+        coralStationAlignPose = new CoralStationAlignPose();
+        coralFloorPose = new CoralFloorPose();
+        coralFloor = new CoralFloor();
+        coralReefPose = new CoralReefPose();
+        coralReef = new CoralReef();
+        coralReefAlignPose = new CoralReefAlignPose();
+        coralReefAligned = new CoralReefAligned();
+        coralScorePose = new CoralScorePose();
+    
+        ejectCoralPose = new EjectCoralPose();
+    }
+
+    @Override
+    public void periodic() {
+
+        Logger.log("current mode", currentMode.getName());
+
+        if (driveMode != null) {
+            switch (driveMode) {
+                case TeleopDrive:
+    
+                    drivetrain.setControl(teleopDrive.withVelocityX(SwerveXSupplier.get() < 0 ? Math.pow(SwerveXSupplier.get() * MaxSpeed, 2) : -Math.pow(SwerveXSupplier.get() * MaxSpeed, 2))
+                                                .withVelocityY(SwerveYSupplier.get() < 0 ? Math.pow(SwerveYSupplier.get() * MaxSpeed, 2) : -Math.pow(SwerveYSupplier.get() * MaxSpeed, 2))
+                                                .withRotationalRate(-SwerveRotationSupplier.get() * MaxAngularRate)
+                    );
+    
+                    break;
+    
+                case FieldCentric:
+    
+                    drivetrain.setControl(fieldCentric.withVelocityX(SwerveXSupplier.get() * MaxSpeed)
+                                                .withVelocityY(SwerveYSupplier.get() * MaxSpeed)
+                                                .withRotationalRate(SwerveRotationSupplier.get() * MaxAngularRate)
+                    );
+                    
+                    break;
+            
+                case RobotCentric:
+    
+                    drivetrain.setControl(robotCentric.withVelocityX(SwerveXSupplier.get() * MaxSpeed)
+                                                    .withVelocityY(SwerveYSupplier.get() * MaxSpeed)
+                                                    .withRotationalRate(SwerveRotationSupplier.get() * MaxAngularRate)
+                    );
+    
+                    break;
+                
+                case Brake:
+    
+                    break;
+    
+                case Auton:
+    
+                    break;
+    
+                default:
+                    break;
+            }
+        }
+        
+    }
+
+    public void setDriveModeCommand(Command newDriveMode) {
+        if (currentDriveMode != null) {
+            currentDriveMode.cancel();
+        }
```

Comment:
Maybe we should check to see if `currentDriveMode == newDriveMode` and do nothing?